plugins/modules/elb_target_group.py

```diff
@@ -566,6 +581,7 @@ def create_or_update_target_group(connection, module):
     params['TargetType'] = target_type
     if target_type != "lambda":
         params['Protocol'] = module.params.get("protocol").upper()
+        params['ProtocolVersion'] = module.params.get('protocol_version')
```

Comment:
Hi @sharvarikhedkar,

Thank you for your PR.

From the CI:

```
2022-09-19 02:29:45.787246 | controller |   File "/tmp/ansible_elb_target_group_payload_gni90h2b/ansible_elb_target_group_payload.zip/ansible_collections/community/aws/plugins/modules/elb_target_group.py", line 822, in create_or_update_target_group
2022-09-19 02:29:45.787254 | controller |   File "/tmp/ansible_elb_target_group_payload_gni90h2b/ansible_elb_target_group_payload.zip/ansible_collections/amazon/aws/plugins/module_utils/modules.py", line 327, in deciding_wrapper
2022-09-19 02:29:45.787260 | controller |     return retrying_wrapper(*args, **kwargs)
2022-09-19 02:29:45.787267 | controller |   File "/tmp/ansible_elb_target_group_payload_gni90h2b/ansible_elb_target_group_payload.zip/ansible_collections/amazon/aws/plugins/module_utils/cloud.py", line 119, in _retry_wrapper
2022-09-19 02:29:45.787274 | controller |     return _retry_func(
2022-09-19 02:29:45.787280 | controller |   File "/tmp/ansible_elb_target_group_payload_gni90h2b/ansible_elb_target_group_payload.zip/ansible_collections/amazon/aws/plugins/module_utils/cloud.py", line 69, in _retry_func
2022-09-19 02:29:45.787287 | controller |     return func()
2022-09-19 02:29:45.787294 | controller |   File "/home/zuul/venv/lib64/python3.9/site-packages/botocore/client.py", line 386, in _api_call
2022-09-19 02:29:45.787300 | controller |     return self._make_api_call(operation_name, kwargs)
2022-09-19 02:29:45.787307 | controller |   File "/home/zuul/venv/lib64/python3.9/site-packages/botocore/client.py", line 677, in _make_api_call
2022-09-19 02:29:45.787313 | controller |     request_dict = self._convert_to_request_dict(
2022-09-19 02:29:45.787320 | controller |   File "/home/zuul/venv/lib64/python3.9/site-packages/botocore/client.py", line 725, in _convert_to_request_dict
2022-09-19 02:29:45.787326 | controller |     request_dict = self._serializer.serialize_to_request(
2022-09-19 02:29:45.787334 | controller |   File "/home/zuul/venv/lib64/python3.9/site-packages/botocore/validate.py", line 319, in serialize_to_request
2022-09-19 02:29:45.787341 | controller |     raise ParamValidationError(report=report.generate_report())
2022-09-19 02:29:45.787347 | controller | botocore.exceptions.ParamValidationError: Parameter validation failed:
2022-09-19 02:29:45.787354 | controller | Invalid type for parameter ProtocolVersion, value: None, type: <class 'NoneType'>, valid types: <class 'str'>
```

`ProtocolVersion` cannot be `None`. It has to be a string. So I would suggest to set the value if it's `!= None`.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.create_target_group

Also, you will need a changelog fragment for a change like this.

Reply:
Hi @goneri, thank you for the review. 
I've added a condition to read the `protocol_version` param only when provided and also added a changelog fragment. Thanks.

Reply:
outdated, I resolve this conversation